src/resolvers/git.cr

```diff
@@ -243,7 +243,34 @@ module Shards
     end
 
     private def origin_changed?
-      (@origin_url ||= capture("git ls-remote --get-url origin").strip) != git_url
+      @origin_url ||= capture("git ls-remote --get-url origin").strip
+      !origins_equal(@origin_url, git_url)
+    end
+
+    # Returns whether origin URLs have matching hosts and paths.
+    #
+    # origins_equal("git@github.com:foo/bar", "https://github.com/foo/bar") # => true
+    protected def origins_equal(origin_1, origin_2)
+      return true if origin_1 == origin_2
+      return false if origin_1.nil? || origin_2.nil?
+
+      re = Regex.union(
+        /[\w\.]+@(?<host>[\w\.]+):\/?(?<path>.*)/,         # git@github.com:foo/bar
+        /(ssh|https?):\/\/(?<host>[^:\/\s]+)\/(?<path>.*)/ # https://github.com/foo/bar
+      )
+
+      match_1 = re.match(origin_1)
+      match_2 = re.match(origin_2)
+
+      return false if match_1.nil? || match_2.nil?
+
+      ["host", "path"].each do |element|
+        if match_1[element] != match_2[element]
+          return false
+        end
+      end
```

Comment:
Please, no `Regex` is needed here, `URI.parse` should be more than enough.

Reply:
See comment below